### PR TITLE
REST high-level client: add cluster get settings API

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
Relates to #27205

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine add to whitelist

File: docs/reference/cluster/get-settings.asciidoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[[cluster-get-settings]]
+== Cluster Get Settings
+
+The cluster get settings API allows to retrieve the cluster wide settings.
```
Comment (nik9000) [inline docs/reference/cluster/get-settings.asciidoc]: :+1:

**Review by nik9000 — APPROVED** [review]
This one LGTM. I'll merge it!

File: server/src/main/java/org/elasticsearch/action/admin/cluster/settings/ClusterGetSettingsResponse.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * This response is specific to the REST client. {@link org.elasticsearch.action.admin.cluster.state.ClusterStateResponse}
+ * is used on the transport layer.
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/settings/ClusterGetSettingsResponse.java]: this is not only specific to the REST client, it's also used in the REST action now? Should this response rather belong to client/high-level (that would require the REST action changes to be reverted though)?

Comment (sohaibiftikhar) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/settings/ClusterGetSettingsResponse.java]: I thought of this. But then I wanted to consolidate the `toXContent` and `fromXContent` in the same place (also helps with tests). It is used in the REST action yes. What I meant by the comment was that it is not streamable and hence not used in the transport layer.

**Pull request merged**